core/jobs/batch_jobs/manual_voice_artist_name_job.py

```diff
@@ -0,0 +1,527 @@
+                # then we should skip the language.
+                if lang_code not in (
+                    latest_content_id_to_voiceover_mapping[content_id]
+                ):
+                    continue
+
+                latest_voiceover_dict_in_given_language = (
+                    latest_content_id_to_voiceover_mapping[
+                        content_id][lang_code])
+
+                # If some old exploration commit contains voiceovers that do
+                # not match the latest voiceover, then we should skip the
+                # iteration.
+                if voiceover_dict != latest_voiceover_dict_in_given_language:
+                    continue
+
+                updated_content_id_to_voiceovers_mapping_tuple[content_id][
+                    lang_code] = (voice_artist_id, voiceover_dict)
+
+                number_of_voiceovers_identified += 1
+
+        return (
+            updated_content_id_to_voiceovers_mapping_tuple,
+            number_of_voiceovers_identified
+        )
+
+    @classmethod
+    def get_exploration_voice_artists_link_model(
+        cls,
+        exploration_model: exp_models.ExplorationModel,
+        snapshot_models: List[exp_models.ExplorationSnapshotContentModel]
+    ) -> voiceover_models.ExplorationVoiceArtistsLinkModel:
+        """The method creates an exploration voice artist link model using the
+        exploration snapshot models for a given exploration model.
+
+        Args:
+            exploration_model: ExplorationModel. The latest version of a
+                curated exploration.
+            snapshot_models: list(ExplorationSnapshotContentModel). A list of
+                exploration snapshot models for every version of the specified
+                curated exploration.
+
+        Returns:
+            ExplorationVoiceArtistsLinkModel. An instance of
+            ExplorationVoiceArtistsLinkModel to find out which voice artists
+            worked on the latest voiceovers by looking at previous versions of
+            exploration snapshots.
+        """
+
+        # The key for sorting is defined separately because of a mypy bug.
+        # A [no-any-return] is thrown if key is defined in the sort()
+        # method instead.
+        # https://github.com/python/mypy/issues/9590
+        k = lambda model: model.id
+        snapshot_models.sort(key=k, reverse=True)
+
+        # Identifying a voiceover means finding the voice artist who contributed
+        # to that voiceover, using the previous versions of exploration
+        # snapshots.
+        total_number_of_voiceovers_to_identify: int = 0
+        total_number_of_voiceovers_identified: int = 0
+
+        # A dictionary mapping all the content IDs and voiceovers of the latest
+        # exploration.
+        latest_content_id_to_voiceover_mapping: Dict[str, Dict[
+            str, voiceover_models.VoiceoverDict]] = {}
+
+        # A dictionary that links all content IDs to tuples containing voice
```

Comment:
Can't really understand this....what exactly does each content ID map to? The current mental model I have from reading this is `content_id: (voiceartist1, voiceartist2, ..., voiceartistn, voiceover1, voiceover2, ..., voiceovern)`